value-fixture/test/org/immutables/fixture/with/WithUnaryOperatorTest.java

```diff
@@ -0,0 +1,43 @@
+package org.immutables.fixture.with;
+
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import static org.immutables.check.Checkers.check;
+
+public class WithUnaryOperatorTest {
+  @Test
+  public void mapAttributes() {
+    ImmutableWithUnaryOperator m = ImmutableWithUnaryOperator.builder()
+        .a(1)
+        .b("b")
+        .addL("x", "y", "z")
+        .o("**")
+        .build();
+
+    ImmutableWithUnaryOperator t = m.mapA(a -> a + 1)
+        .mapB(b -> "[" + b + "]")
+        .mapL(String::toUpperCase)
+        .mapO(o -> o.replace('*', '_'));
```

Comment:
does it make sense to have `flatMap` on optionals ?

Reply:
Maybe, idk. Because how this feature is activated (by a naming template), I see that flatmap will require separate style flag to enable, `Style.withFlatmap` for example.
Also it would be fair for collections to  also have flatmap, but it will probably need 2 overloads,  like  `Function<? super T,  ? extends Iterable<? extends T>>` and `Function<? super T,  ? extends Stream<? extends T>>`.
BTW (DYK)  I've just learned couple of days ago that you can return `null` in the map function (JDK Optional)

```java
assert Optional.of("X").map(x -> null).isEmpty(); // no NPE
```

I think if there would be any demand for flatMap, which I see as a separate flag, I can implement flatMap for optional / collections. Otherwise, I think, Encodings should be used to customize any other specific container or collection.
I've already planned to add smth simple for `map`, there were at least couple of use cases where this is useful for us. Flatmap is more exotic in that sense, that while it's logical to have, we have never needed it on the surface of immutable object.